docs/src/tutorial.md

```diff
@@ -4,49 +4,45 @@
-5. [Optional] Visualize the model and animate its time evolution. This can help checking that the model behaves as expected and there aren't any mistakes, or can be used in making figures for a paper/presentation.
+2. Define the agent type (or types, for mixed models) that will populate the ABM. Agent types are standard Julia `mutable struct` that are created with [`@agent`](@ref). The types must contain some mandatory fields, which is ensured by using [`@agent`](@ref). The remaining fields of the agent type are up to user's choice.
+3. The created agent type, the chosen space, optional additional model level properties, and other simulation tuning properties like schedulers or random number generators, are given to our universal structure [`AgentBasedModel`](@ref). This instance defines the model within an Agents.jl simulation.
+4. Provide functions that govern the time evolution of the ABM. A user can provide an agent-stepping function, that acts on each agent one by one, and/or a model-stepping function, that steps the entire model as a whole. These functions are standard Julia functions that take advantage of the Agents.jl [API](@ref). Once these functions are created, they are simply passed to [`step!`](@ref) to evolve the model.
+5. _(Optional)_ Visualize the model and animate its time evolution. This can help checking that the model behaves as expected and there aren't any mistakes, or can be used in making figures for a paper/presentation.
 6. Collect data. To do this, specify which data should be collected, by providing one standard Julia `Vector` of data-to-collect for agents, for example `[:mood, :wealth]`, and another one for the model. The agent data names are given as the keyword `adata` and the model as keyword `mdata` to the function [`run!`](@ref). This function outputs collected data in the form of a `DataFrame`.
 
 If you're planning of running massive simulations, it might be worth having a look at the [Performance Tips](@ref) after familiarizing yourself with Agents.jl.
 
 
 ## [1. The space](@id Space)
 Agents.jl offers several possibilities for the space the agents live in.
-In addition, it is straightforward to implement a fundamentally new type of space, see [Developer Docs](@ref).
+In addition, it is straightforward to implement a fundamentally new type of space, see [Creating a new space type](@ref).
 
-The available spaces are:
+The available spaces are listed in the [Available spaces](@ref) part of the API.
+An example of a space is [`OpenStreetMapSpace`](@ref). It is based on Open Street Map, where agents are confined to move along streets of the map, using real-world meter values for the length of each street.
```

Comment:
> using real-world meter values for the length of each street

This is not true. The unit of length in OSMSpace is arbitrary, and can be length, time to cross, or another quantity altogether.